server/lib/models/matchmaking-times.ts

```diff
@@ -0,0 +1,200 @@
+import db from '../db'
+import sql from 'sql-template-strings'
+import { MatchmakingType } from '../../../common/matchmaking'
+
+export interface MatchmakingTime {
+  id: string
+  type: MatchmakingType
+  startDate: Date
+  enabled: boolean
+}
+
+function convertFromDb(props: {
+  /* eslint-disable camelcase */
+  id: string
+  matchmaking_type: MatchmakingType
+  start_date: Date
+  enabled: boolean
+  /* eslint-enable camelcase */
+}): MatchmakingTime {
+  return {
+    id: props.id,
+    type: props.matchmaking_type,
+    startDate: props.start_date,
+    enabled: props.enabled,
+  }
+}
+
+export async function getCurrentMatchmakingTime(
+  matchmakingType: MatchmakingType,
+): Promise<MatchmakingTime | null> {
+  const query = sql`
+    SELECT *
+    FROM matchmaking_times
+    WHERE matchmaking_type = ${matchmakingType} AND start_date <= ${new Date()}
+    ORDER BY start_date DESC
+    LIMIT 1;
+  `
+
+  const { client, done } = await db()
+  try {
+    const result = await client.query(query)
+    return result.rows.length > 0 ? convertFromDb(result.rows[0]) : null
+  } finally {
+    done()
+  }
+}
+
+async function getTotalTimes(
+  matchmakingType: MatchmakingType,
+  date = new Date(),
+  isFuture = true,
+): Promise<number> {
+  const query = sql`
+    SELECT COUNT(id)
```

Comment:
Generally I think you're better off just doing `COUNT(*)`. I assume you're doing `COUNT(id)` under an assumption that it will be faster, since it names less columns, but `COUNT(*)` is actually parsed as a specific thing and doesn't require the DB to look at every row, whereas your query *does* require looking at the rows.